modules/imaging_browser/php/imaging_session_controlpanel.class.inc

```diff
@@ -96,13 +96,20 @@ class Imaging_Session_ControlPanel
             $params['PCandID'] = $timePoint->getCandID();
             $params['PVL']     = $timePoint->getVisitLabel();
         }
-        $tarchiveIDs = $DB->pselect(
-            "SELECT TarchiveID 
+        // To enable DICOM download: Add pair: tarchiveID, ArchiveLocation
+        $qresult = $DB->pselect(
+            "SELECT TarchiveID, ArchiveLocation 
             FROM tarchive 
             WHERE PatientName LIKE $ID",
             $params
         );
-        $subjectData['tarchiveids'] = $tarchiveIDs;
+
+        $tarIDToTarLoc =array();
+        foreach ($qresult as $k=>$v) {
+            $tarchiveID = $v['TarchiveID'];
+            $tarIDToTarLoc[$tarchiveID] =$v['ArchiveLocation'];
+        }
```

Comment:
suggest using array_values() instead of foreach. Prevent unused variable $k.

Reply:
removed unused `$k`. thanks!